### ISSUE TITLE
Main Search, add debounce to reduce the number of Elastic calls

### ISSUE THREAD

**Issue by RobOatesHistoricEngland** [archesproject/arches#9820]
Search is excessively making calls to /search/terms endpoint when two or more characters are input. For example, if `New Covent Garden` search term is entered, a total of 16 Elastic searches will be made.

Introducing a debounce of 1000ms will reduce this to a single Elastic search - providing each key-press is made within one second of each another.

Select2 provides a debounce mechanism via `quietMillis`. Adding this parameter to `term-search.js` should do the trick:

```javascript
var searchbox = $(el).select2({
    dropdownCssClass: 'resource_search_widget_dropdown',
    multiple: true,
    minimumInputLength: 2,
    ajax: {
        url: arches.urls.search_terms,
        dataType: 'json',
        quietMillis: 1000, <== New code to insert
        data: function(term, page) {
            return {
                q: term, // search term
                page_limit: 30
            };
        },
```

**Comment by RobOatesHistoricEngland** [comment]
@chiatt @apeters Any feedback, or are we good to go? Thanks, Rob.

**Comment by chiatt** [comment]
Hi @RobOatesHistoricEngland - That seems like a nice change to me. I suspect 1000ms might be a bit long. It would nice to minimize any lag a user might experience after they are done typing. Have you experimented with a smaller number, maybe 250 or 500?

**Comment by RobOatesHistoricEngland** [comment]
Hi @chiatt, I did try 500ms and played devils advocate as a slow typer which generated more ES requests, but still far fewer than not having a debounce at all.

I wonder if something like 500ms should be the default value and have this overridable in settings_local?

**Comment by apeters** [comment]
@RobOatesHistoricEngland I would think that 500ms would be a nice happy medium.

**Comment by RobOatesHistoricEngland** [comment]
Thank you, gentlemen. 500ms is the winner.

**Comment by jacobtylerwalls** [comment]
Fixed in #9946
